test/ngMock/angular-mocksSpec.js

```diff
@@ -2505,6 +2505,44 @@ describe('`afterEach` clean-up', function() {
     });
   });
 
+  describe('ErrorAddingDeclarationLocationStack', function() {
+    it('should be caught by jasmine\'s toThrowError', function() {
+      //create function that causes an error
+      function causeNgMocksError() {
+        module(function($provide) {
+          //cause $provide.factory to include an error
+          $provide.factory('causeError', function() {
+            //Say for example, some mock module caused an error when instantiated
+            throw new Error("Error!");
+          });
+        });
+
+        //call the injector on 'causeError' to make sure the provider is called
+        var causeError;
+        inject(function(_causeError_) {
+          causeError = _causeError_;
+        });
+      }
+
+      //The following is a simplified implementation of future Jasmine's .toThrowError()
```

Comment:
`toThrowError()` was introduced in Jasmine 2.0. It uses `errorObject instanceof Error` to check if the thrown object is an Error Object (for matching strings/regex against their `.message` property). Angular's build environment uses now >2-year-old Jasmine 1.3 for its own tests, so I had to rig a basic function for how Jasmine's `toThrowError()` fails when running this code.